packages/typespec-test/test/contentsafety_modular/spec/models.tsp

```diff
@@ -0,0 +1,195 @@
+enum ImageCategory {
+  Hate,
+  SelfHarm,
+  Sexual,
+  Violence,
+}
+
+@doc("The type of text analysis output.")
+enum AnalyzeTextOutputType {
+  @doc("Output severities in four levels, the value could be 0,2,4,6.")
+  FourLevels,
+
+  @doc("Output severities in four levels, the value could be 0,1,2,3,4,5,6,7.")
+  EightLevels,
+}
+
+@doc("The type of image analysis output.")
+enum AnalyzeImageOutputType {
+  @doc("Output severities in four levels, the value could be 0,2,4,6.")
+  FourLevels,
+}
+
+@doc("The analysis request of the text.")
+model AnalyzeTextOptions {
+  // @doc("The text needs to be scanned. We support at most 1000 characters (unicode code points) in text of one request.")
+  // @maxLength(1000)
+  @doc("The text needs to be scanned. We support at most 10k Unicode characters (unicode code points) in text of one request.")
+  @maxLength(10000)
+  text: string;
+
+  @doc("The categories will be analyzed. If not assigned, a default set of the categories' analysis results will be returned.")
+  categories?: TextCategory[];
+
+  @doc("The names of blocklists.")
+  blocklistNames?: string[];
+
+  @doc("When set to true, further analyses of harmful content will not be performed in cases where blocklists are hit. When set to false, all analyses of harmful content will be performed, whether or not blocklists are hit.")
+  breakByBlocklists?: boolean;
+
+  @doc("The type of text analysis output. If not assigned, the default value is \"FourLevels\".")
+  outputType?: AnalyzeTextOutputType = AnalyzeTextOutputType.FourLevels;
+}
+
+@doc("The analysis response of the text")
+model AnalyzeTextResult {
+  @doc("The details of blocklist match.")
+  blocklistsMatchResults?: TextBlocklistMatchResult[];
+
+  @doc("Analysis result for categories.")
+  analyzeResults: TextAnalyzeSeverityResult[];
+}
+
+@doc("The result of blocklist match.")
+model TextBlocklistMatchResult {
+  @doc("The name of matched blocklist.")
+  @maxLength(64)
+  blocklistName: string;
+
+  @doc("The id of matched item.")
+  @maxLength(64)
+  blockItemId: string;
+
+  @doc("The content of matched item.")
+  @maxLength(128)
+  blockItemText: string;
+}
+
+@doc("Text analysis result.")
+model TextAnalyzeSeverityResult {
+  @doc("The text category.")
+  category: TextCategory;
+
+  @doc("This field is decided by outputType in request, if choose \"FourLevels\", the value could be 0,2,4,6. The higher the severity of input content, the larger this value is.")
+  severity?: int32;
+}
+
+@doc("The analysis request of the image.")
+model AnalyzeImageOptions {
+  @doc("The image needs to be analyzed.")
+  image: ImageData;
+
+  @doc("The categories will be analyzed. If not assigned, a default set of the categories' analysis results will be returned.")
+  categories?: ImageCategory[];
+
+  @doc("The type of image analysis output. If not assigned, the default value is \"FourLevels\".")
+  outputType?: AnalyzeImageOutputType = AnalyzeImageOutputType.FourLevels;
```

Comment:
Here in their TypeSpec, they have specified the default value for outputType is FourLevels, but in our generated code, we don't reflect this in either modular layer or rlc layer.